build/migration-guide.md

```diff
@@ -0,0 +1,109 @@
+
+## Validator Registration Tool
+
+The Validator Registration tool facilitates the registration process for users migrating from PoW to PoS during the pre-registration phase. Users register within the PoW chain. The Validator Registration tool operates in two distinct modes:
+
+- Address and corresponding keys generation
+- Validator registration via transactions
+
+ First, you need to clone the [core-js repository](https://github.com/nimiq/core-js?tab=readme-ov-file#quickstart) and follow the instructions.
+
+### Step 1: Generate the keys
+
+Execute the script without parameters to generate the validator address, signing key, and voting key:
+
+```shell
+node validator-registration.js
+```
+
+The tool generates the keys and pastes them into the `validator-keys.json` file. The screenshot demonstrates the output of the script.
+
+<img class="object-contain max-h-[max(80vh,220px)]" src="/assets/images/migration/migration.png" alt="skip block struct" />
+
+<aside>
+💡 Save the private key securely! There is no recovery mechanism for lost private keys. Once lost, access to your validator and NIM may be permanently lost.
+
+</aside>
+
+### Step 2: Fund your address
+
+To start the validator registration process, fund the address you just generated to cover the nominal transaction fees of 1 Luna each (you need 6 Lunas) associated with the validator generation process. You can use any wallet to fund this address.
+
+### Step 3: Execute the Validator Registration Script
+
+::: details Testnet {open}
+
+For testing this in the PoW testnet, you must specify the network argument as follows:
+
+```shell
+node validator-registration.js --validator validator-keys.json --network test
+```
+
+:::
+
+::: details Mainnet
+
+Run the following script with your address and public keys as parameters:
+
+```shell
+node validator-registration.js --validator validator-keys.json
+```
+
+:::
+
+The script will connect the tool to the Nimiq PoW chain. Upon consensus, the tool creates the validator registration transaction, signs it, and sends it to the network, effectively generating the validator. Note that this transaction is a bundle of 6, and each one has a nominal fee of 1 Luna. Please ensure you have funded your address—as instructed in the previous step—as the transaction fees are paid from this address.
+
+For detailed guidance through the scripts, run `node validator-registration.js --help`. This will print out the following usage description.
+
+### Step 4:  Deposit Payment and Commit
+
+The final transaction involves committing to the registration and paying the validator's deposit of 100 000 NIM. This transaction can be sent from any address. Despite the address you use to pay the validator deposit, **you must include your validator address** in the transaction’s “public message” field so we can later identify your validator.
```

Comment:
you must include your validator address in _human-readable_ format in the transaction's "public message"...